connectors/source.py

```diff
@@ -106,19 +106,27 @@ def value(self, value):
         self._value = value
 
     def _convert(self, value, type_):
-        if not isinstance(value, str):
-            # we won't convert the value if it's not a str
-            return value
-
-        if type_ == "int":
-            return int(value)
-        elif type_ == "float":
-            return float(value)
-        elif type_ == "bool":
-            return value.lower() in ("y", "yes", "true", "1")
-        elif type_ == "list":
-            return [item.strip() for item in value.split(",")]
-        return value
+        match type_:
+            case "int":
+                if isinstance(value, int):
+                    return value
+                return int(value) if value else None
+            case "float":
+                if isinstance(value, float):
+                    return value
+                return float(value) if value else None
+            case "bool":
+                if isinstance(value, bool):
+                    return value
+                return bool(value) if value is not None else None
+            case "list":
+                if isinstance(value, list):
+                    return value
+                if isinstance(value, str):
+                    return [item.strip() for item in value.split(",")] if value else []
+                return [value] if value is not None else []
+            case _:  # str
+                return str(value) if value else ""
```

Comment:
This seems can be simplified to:

```python
from pydoc import locate

casted_type = locate(type_)

if isinstance(value, casted_type):
    return value

return casted_type(value) if value else None
```

But you need to do some special handling like `strip()`, etc..

Reply:
I will try to simplify it, some of these cases could indeed be consolidated into a default case.
But yeah we need to handle lists differently and there's some subtle differences in handling `None` between numbers,strings, etc.

Reply:
@wangch079 I've simplified the logic using `locate` but it required me to add some `pyright:ignore` comments. Let me know if that's okay. It's caused by pyright being unsure of what class exactly `cast_type` becomes. I added a type check clause earlier in the function so it shouldn't be an issue.

Specifics:
- `if isinstance(value, cast_type)` -> `Object of type "object" is not callable`
- `cast_type(value)` -> `Argument of type "object" cannot be assigned to parameter "__class_or_tuple" of type "_ClassInfo" in function "isinstance"`